app/src/main/java/com/github/se/cyrcle/model/parking/ParkingRepositoryFirestore.kt

```diff
@@ -0,0 +1,167 @@
+
+  override fun getNewUid(): String {
+    return db.collection(collectionPath).document().id
+  }
+
+  override fun init(onSuccess: () -> Unit) {
+    Firebase.auth.addAuthStateListener {
+      if (it.currentUser != null) {
+        onSuccess()
+      }
+    }
+  }
+
+  override fun getParkings(onSuccess: (List<Parking>) -> Unit, onFailure: (Exception) -> Unit) {
+    Log.d("ParkingRepositoryFirestore", "getParkings")
+    db.collection(collectionPath).get().addOnCompleteListener { task ->
+      if (task.isSuccessful) {
+        val parkings =
+            task.result?.mapNotNull { document -> deserializeParking(document.data) } ?: emptyList()
+        onSuccess(parkings)
+      } else {
+        task.exception?.let { e ->
+          Log.e("ParkingRepositoryFirestore", "Error getting documents", e)
+          onFailure(e)
+        }
+      }
+    }
+  }
+
+  override fun getParkingById(
+      id: String,
+      onSuccess: (Parking) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .document(id)
+        .get()
+        .addOnSuccessListener { document ->
+          val parking = document.data?.let { deserializeParking(it) }
+          if (parking != null) {
+            onSuccess(parking)
+          } else {
+            onFailure(Exception("Parking not found"))
+          }
+        }
+        .addOnFailureListener { onFailure(it) }
+  }
+
+  override fun getParkingsBetween(
+      start: Point,
+      end: Point,
+      onSuccess: (List<Parking>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    if (start.latitude > end.latitude || start.longitude > end.longitude) {
+      onFailure(Exception("Invalid range"))
+      return
+    }
+    db.collection(collectionPath)
+        .whereGreaterThanOrEqualTo("location.center.latitude", start.latitude)
+        .whereLessThanOrEqualTo("location.center.latitude", end.latitude)
+        .whereGreaterThanOrEqualTo("location.center.longitude", start.longitude)
+        .whereLessThanOrEqualTo("location.center.longitude", end.longitude)
+        .get()
+        .addOnCompleteListener { task ->
+          if (task.isSuccessful) {
+            val parkings =
+                task.result?.mapNotNull { document -> deserializeParking(document.data) }
+                    ?: emptyList()
+            onSuccess(parkings)
+          } else {
+            task.exception?.let { e ->
+              Log.e("ParkingRepositoryFirestore", "Error getting documents", e)
+              onFailure(e)
+            }
+          }
+        }
+  }
+
+  override fun getKClosestParkings(
+      location: Point,
+      k: Int,
+      onSuccess: (List<Parking>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    // variables for radius search
+    val initialRadius = 0.01
+    val maxRadius = 0.5
+    val step = 0.01
+
+    var currentRadius = initialRadius
+    var parkings = emptyList<Parking>()
+
+    fun getMoreParkings() {
```

Comment:
Documentation for this function may be useful since not defined in the interface

Reply:
I judged that documentation was not needed given that this is a local function. But it is true that for code maintenance it could have been helpful